### DISCUSSION TITLE
Skill-up Magento Development Team from local env, to Magento Framework, coding best practices and Solution Arch

### DISCUSSION THREAD

**Discussion by beseiso** [markshust/docker-magento#872] [Q&A]
My team and I are receiving MacBook Pros with M2 chips with max RAM.

We have the option of direct install, Docker, Warden, etc.

Our intention is to enable Devs to have multiple environments simultaneously, that way, they can support production bug fixes by having mirror of Prod code, new features in a Dev instance, and maybe a sandbox to Tinker/Innovate in, etc.

What recommendations does the Magento community have for local development environments: Direct install, Docker, or Warden+Docker?

What recommendations for IDE? Visual Studio Code, Visual Studio, PHP Storm, Sublime, etc.?

What recommendations to skill up the Magento Developers in streamlined manner from Novice to Best Practices?

In advance, appreciate Experts who are seasoned Magento Developers, to provide insights.

Thank you,
JSB

**Comment by markshust** [comment]
I very much recommend this docker-magento setup 🤓 It's by far the most stable & secure dev env out there, and I've onboarded hundreds of teams onto it.

IDE: PhpStorm. Nothing comes close.

Skill up? Check out [M.academy](https://m.academy)! My entire livelihood revolves around it. I don't check these discussions too often, so if you have questions about training be sure to reach out through the contact channels on the M.academy website. You can also read the case study of our DEVTEAM onboarded their entire team using M.academy at https://m.academy/case-studies/dev-team/

Hope this was helpful!
